src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/CompileFails.cs

```diff
@@ -418,5 +419,75 @@ partial interface {|#0:IFoo2|}<T>
 
             await VerifyComInterfaceGenerator.VerifySourceGeneratorAsync(source, expectedDiagnostic);
         }
+
+        [Fact]
+        public async Task VerifyComInterfaceInheritingFromComInterfaceInOtherAssemblyReportsDiagnostic()
+        {
+            string additionalSource = $$"""
+                using System.Runtime.InteropServices;
+                using System.Runtime.InteropServices.Marshalling;
+       
+                [GeneratedComInterface]
+                [Guid("9D3FD745-3C90-4C10-B140-FAFB01E3541D")]
+                public partial interface I
+                {
+                    void Method();
+                }
+                """;
+
+            string source = $$"""
+                using System.Runtime.InteropServices;
+                using System.Runtime.InteropServices.Marshalling;
+
+                [GeneratedComInterface]
+                [Guid("0DB41042-0255-4CDD-B73A-9C5D5F31303D")]
+                partial interface {|#0:J|} : I
+                {
+                    void MethodA();
+                }
+                """;
+
+            var test = new VerifyComInterfaceGenerator.Test(referenceAncillaryInterop: false)
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        ("Source.cs", source)
+                    },
+                    AdditionalProjects =
+                    {
+                        ["Other"] =
+                        {
+                            Sources =
+                            {
+                                ("Other.cs", additionalSource)
+                            },
+                        },
+                    },
+                    AdditionalProjectReferences =
+                    {
+                        "Other"
+                    }
+                },
+                TestBehaviors = TestBehaviors.SkipGeneratedSourcesCheck | TestBehaviors.SkipGeneratedCodeCheck,
+            };
+            test.TestState.AdditionalProjects["Other"].AdditionalReferences.AddRange(test.TestState.AdditionalReferences);
+
+            test.ExpectedDiagnostics.Add(
+                VerifyComInterfaceGenerator
+                    .Diagnostic(GeneratorDiagnostics.BaseInterfaceIsNotGenerated)
+                    .WithLocation(0)
+                    .WithArguments("J", "I"));
+
+            // The Roslyn SDK doesn't apply the compilation options from CreateCompilationOptions to AdditionalProjects-based projects.
+            test.SolutionTransforms.Add((sln, _) =>
+            {
+                var additionalProject = sln.Projects.First(proj => proj.Name == "Other");
+                return additionalProject.WithCompilationOptions(new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, allowUnsafe: true)).Solution;
+            });
```

Comment:
Do you think it's worth it to make a new Test subclass for tests with AdditionProjects?

Reply:
No, I think doing it manually for the cases where we need it is okay. It's not a lot of packages and it's easy to mess with.

Also I'll probably fix this issue in the Roslyn-SDK at some point and then we can get rid of the solution transform.